easybuild/easyconfigs/c/CVX/CVX-2.2-MATLAB-2023a.eb

```diff
@@ -0,0 +1,36 @@
+easyblock = 'Tarball'
+
+name = 'CVX'
+version = '2.2'
+_matlab_ver = '2023a'
+versionsuffix = '-MATLAB-%s' % _matlab_ver
+
+homepage = 'https://cvxr.com/cvx/'
+description = """CVX is a Matlab-based modeling system for convex optimization.
+ CVX turns Matlab into a modeling language, allowing constraints and objectives
+ to be specified using standard Matlab expression syntax.
+"""
+
+toolchain = SYSTEM
+
+source_urls = ['https://web.cvxr.com/cvx/']
+sources = ['%(namelower)s-a64.tar.gz']
+checksums = ['16e4622c80f2bf63152aaee59db4fe42afa4d2282179e5d216358953c7f9ea4d']
+
+dependencies = [
+    ('MATLAB', _matlab_ver),
+]
+
+sanity_check_paths = {
+    'files': [],
+    'dirs': ['lib', 'commands', 'sedumi', 'sdpt3'],
+}
+
+modloadmsg = "IMPORTANT: You need executing `cvx_setup` in the first time."
```

Comment:
This message is not clear. What do you mean by it?

Reply:
To use cvx in matlab, it requires executing `cvx_setup` at least once in matlab. Otherwise, cvx cannot work properly.

```matlab
%%% new clean matlab environment
>> cvx_begin
Dot indexing is not supported for variables of this type.

Error in cvx_global (line 76)
osolvers = cvx___.solvers.list;

Error in cvxprob (line 4)
cvx_global

Error in cvx_begin (line 41)
assignin( 'caller', 'cvx_problem', cvxprob( varargin{:} ) );

>> cvx_setup

---------------------------------------------------------------------------
CVX: Software for Disciplined Convex Programming       (c)2014 CVX Research
Version 2.2, Build 1148 (62bfcca)                  Tue Jan 28 00:51:35 2020
---------------------------------------------------------------------------
Installation info:
    Path: /apps/Test2/software/CVX/2.2-MATLAB-2023a
    MATLAB version: 9.14 (R2023a)
    OS: Linux amd64 version 4.18.0-513.11.1.el8_9.0.1.x86_64
    Java version: 1.8.0_202
Verfying CVX directory contents:
    WARNING: The following files/directories are missing:
        /apps/Test2/software/CVX/2.2-MATLAB-2023a/sedumi/.travis.yml
    These omissions may prevent CVX from operating properly.
    WARNING: The following extra files/directories were found:
        /apps/Test2/software/CVX/2.2-MATLAB-2023a/easybuild/ + 8 files, 3 subdirectories
        /apps/Test2/software/CVX/2.2-MATLAB-2023a/lib64/ + 159 files, 9 subdirectories
    These files may alter the behavior of CVX in unsupported ways.
Preferences: none found; defaults loaded.
License host:
    Username: chiajung
    Host ID: 98039b0f34e3 (ens1f1np1.10,129.16.125.38)
Installed license:
    No license installed.
No valid licenses found.
    <a href="http://cvxr.com/cvx/academic?c_act=fill&c_unm=chiajung&c_hid=98039b0f34e3
">Click here</a> to fill out an academic license request
    for the username and first hostid listed above.
---------------------------------------------------------------------------
Setting CVX paths...already set!
Searching for solvers...5 shims found.
2 solvers initialized (* = default):
 *  SDPT3    4.0        {cvx}/sdpt3
    SeDuMi   1.3.4      {cvx}/sedumi
3 solvers skipped:
    GLPK
        Could not find a GLPK installation.
    Gurobi   unknown    {cvx}/gurobi/a64
        No valid Gurobi license was found.
    Mosek    9.1.9      {cvx}/mosek/a64
        Using MOSEK with CVX requires an academic MOSEK license
        or a MOSEK-enabled CVX Professional license.
Saving updated preferences...done.
Testing with a simple model...done!
---------------------------------------------------------------------------
To change the default solver, type "cvx_solver <solver_name>".
To save this change for future sessions, type "cvx_save_prefs".
Please consult the users' guide for more information.
---------------------------------------------------------------------------

>> cvx_begin
%%% pass
```

Reply:
Ok then perhaps something like:
"IMPORTANT: You need to run `cvx_setup` once inside MATLAB before using CVX."

Reply:
Fixed. Thanks!